datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -378,6 +376,13 @@ private DatastreamRuntimeException generateSendFailure(Exception exception, Data
     }
   }
 
+  /*
+   * Kafka producer throws IllegalStateException when the close is in progress. So, instead of attempting to close
```

Comment:
All actual exceptions thrown by the open source producer is wrapped by an `IllegalStateException` in our apache li open source Kafka clients. So interrupt and timeout exceptions will also be thrown as `IllegalStateException`.

Reply:
@vmaheshw if this is the case, then we might want to distinguish between `IllegalStateException`s caused by a flush attempt during producer close and interrupts. Otherwise the `InterruptedException` will not be rethrown.

Reply:
Can you fix the comment to indicate that `IllegalStateException` is thrown for all exceptions and that you've added special handling if close is in progress? Perhaps move this comment closer to the actual exception handling code

Reply:
done